evaluator/builtin_time.go

```diff
@@ -673,6 +675,326 @@ func builtinFromUnixTime(args []types.Datum, _ context.Context) (d types.Datum,
+			return input[i:]
+		}
+	}
+	return ""
+}
+
+var weekdayAbbrev = map[string]time.Weekday{
+	"Sun": time.Sunday,
+	"Mon": time.Monday,
+	"Tue": time.Tuesday,
+	"Wed": time.Wednesday,
+	"Thu": time.Tuesday,
+	"Fri": time.Friday,
+	"Sat": time.Saturday,
+}
+
+var monthAbbrev = map[string]time.Month{
+	"Jan": time.January,
+	"Feb": time.February,
+	"Mar": time.March,
+	"Apr": time.April,
+	"May": time.May,
+	"Jun": time.June,
+	"Jul": time.July,
+	"Aug": time.August,
+	"Sep": time.September,
+	"Oct": time.October,
+	"Nov": time.November,
+	"Dec": time.December,
+}
+
+type dateFormatParser func(t *time.Time, date string) (remain string, succ bool)
+
+var dateFormatParserTable = map[string]dateFormatParser{
+	"%a": abbreviatedWeekday,
+	"%b": abbreviatedMonth,
+	"%c": monthNumeric,
+	"%D": dayOfMonthWithSuffix,
+	"%Y": yearNumericFourDigits,
+	"%m": monthNumericTwoDigits,
+	"%d": dayOfMonthNumericTwoDigits,
+	"%H": hour24TwoDigits,
+	"%i": minutesNumeric,
+	"%s": secondsNumeric,
+}
+
+func matchDateWithToken(t *time.Time, date string, token string) (remain string, succ bool) {
+	if parse, ok := dateFormatParserTable[token]; ok {
+		return parse(t, date)
+	}
+
+	if strings.HasPrefix(date, token) {
+		return date[len(token):], true
+	}
+	return date, false
+}
+
+func parseTwoDigits(input string) (int, bool) {
+	if len(input) < 2 {
+		return 0, false
+	}
+
+	v, err := strconv.ParseUint(input[:2], 10, 64)
+	if err != nil {
+		return int(v), false
+	}
+	return int(v), true
+}
+
+func hour24TwoDigits(t *time.Time, input string) (string, bool) {
+	v, succ := parseTwoDigits(input)
+	if !succ || v >= 24 {
+		return input, false
+	}
+	timeSetHour(t, v)
+	return input[2:], true
+}
+
+func secondsNumeric(t *time.Time, input string) (string, bool) {
+	v, succ := parseTwoDigits(input)
+	if !succ || v >= 60 {
+		return input, false
+	}
+	timeSetSecond(t, v)
+	return input[2:], true
+}
+
+func minutesNumeric(t *time.Time, input string) (string, bool) {
+	v, succ := parseTwoDigits(input)
+	if !succ || v >= 60 {
+		return input, false
+	}
+	timeSetMinute(t, v)
+	return input[2:], true
+}
+
+func dayOfMonthNumericTwoDigits(t *time.Time, input string) (string, bool) {
+	v, succ := parseTwoDigits(input)
+	if !succ || v >= 32 {
+		return input, false
+	}
+	timeSetDay(t, v)
+	return input[2:], true
+}
+
+func yearNumericFourDigits(t *time.Time, input string) (string, bool) {
+	if len(input) < 4 {
+		return input, false
+	}
+
+	v, err := strconv.ParseUint(input[:4], 10, 64)
+	if err != nil {
+		return input, false
+	}
+	timeSetYear(t, int(v))
+	return input[4:], true
+}
+
+func monthNumericTwoDigits(t *time.Time, input string) (string, bool) {
+	v, succ := parseTwoDigits(input)
+	if !succ || v > 12 {
+		return input, false
+	}
+
+	timeSetMonth(t, time.Month(v))
+	return input[2:], true
+}
+
+func abbreviatedWeekday(t *time.Time, input string) (string, bool) {
+	if len(input) >= 3 {
+		dayName := input[:3]
+		if _, ok := weekdayAbbrev[dayName]; ok {
```

Comment:
I'll confirm the behavior of MySQL, in the next PRs
Currently this function is not really supported, as our time representation is not compatible with mysql.